### ISSUE TITLE
ClusterSecretStore failing with x509: certificate signed by unknown authority even after specifying a CA Bundle for HashiCorp Vault

### ISSUE THREAD

**Issue by ramakrishnateja** [external-secrets/external-secrets#2008]
**Describe the solution you'd like**
Would like to have a ability to authenticate to vault with a CABundle or atleast an option to bypass the tls verification

** Yaml configuration used **
apiVersion: external-secrets.io/v1beta1
kind: ClusterSecretStore
metadata:
  name: vault-backend
spec:
  provider:
    vault:
      server: "https://vault-us.xxxinternal.com"
      path: "pwclabs"
      version: "v2"
      auth:
        appRole:
          path: "approle"
          roleId: "xxxxxxxx-xxxxx-xxxx-xxxxxxxx"
          secretRef:
            name: "infra-vault-approle-secret"
            namespace: "infra"
            key: "secret-id"
      caBundle: "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURJRENDQWdpZ0F3SUJBZ0lRVUZvQkFhM2ZPQ1U4VHNRYWpGdGZLREFOQmdrcWhraUc5dzBCQVFzRkFEQXEKTVNnd0pnWURWUVFERXg5amIzVmphR0poYzJVdGIzQmxjbUYwYjNJdFlXUnRhWE56YVc5dUlFTkJNQjRYRFRFNQpNRFF5TlRFM01USXlPRm9YRFRJNU1EUXlNakUzTVRJeU9Gb3dLakVvTUNZR0ExVUVBeE1mWTI5MVkyaGlZWE5sCkxXOXdaWEpoZEc5eUxXRmtiV2x6YzJsdmJpQkRRVENDQVNJd0RRWUpLb1pJaHZjTkFRRUJCUUFEZ2dFUEFEQ0MKQVFvQ2dnRUJBSjdPMVNxbWhBb05CdURBdzBRc2JwS3pwa01LVWVybHRhV3A1VklPVzlnMzdCZ1BrWkMvK3hCSQpOcmoraUdOOC9nM3dkVEFxU2JFaGtKUVhxRWY3OFlhM2hUNlVqdVkwRTdQbWN4TzUrdTZDNW1zL3VBZVRoSW42CnlyRE54ajNOSG12bmUrQ1FzZW5zWWhEbjhDU1QwUnZhU3RZbFlaZHBidWVDc0dXSlJYOVBPSG5xSEpPY3VKOWsKZVI2d0drREFFQWI3RDZ5N3ZHdjQ3MUR3TktxRzZ4ZlhDM3c3eUExMEczSWZMRU44Rm02bWFCaTZZbVdyTHpuMwoyajNkTm5FbENYbUtRZHNhMTZjZjZYd1hUbTNlMFNTSXFQa0VJbDhsZUYrcXp4blVoRlcxSFM1S1AyejJaTDArCmsyUmlxMWlPQ1NId0RNMWdrVjdRK0lNZDZIK25sZ0VDQXdFQUFhTkNNRUF3RGdZRFZSMFBBUUgvQkFRREFnRUcKTUE4R0ExVWRFd0VCL3dRRk1BTUJBZjh3SFFZRFZSME9CQllFRkF3ajZSYkEwdkdHd3ZGcmtPNnJxaEl3VnV0MQpNQTBHQ1NxR1NJYjNEUUVCQ3dVQUE0SUJBUUFZREJ3ejNYTGNYemhkYzZBSys0a3RFRzZ5VU5XUWZvblNXSi9iCkw0NjRGVUcyVmNXMEpnczlNM25adm9SUldlU0Nwci9ZejlzazNGUlVabkhkdFZ2S1ZDaEFGNWk2dnBUd0tmMEQKRklKdGtSa1JWMllIdDlWb3BzeWxhWGljSmk4WTh1V1J2dFNTVHRUR3NHeFV3eGhwR1FraVgvNkVYRHBCUjlYbAo2VTF5TUxnL0I3b2phR2t4ODV3MmNyeitIMWVEcVhzTVl6RUJpK2JTd3FZWU9yTDA4djk5OFNsaXZ5V0lDL2VWCk1wTDY0RUNVWmM3OU5zc1BWVWxQUy9JcytzRmJEUDlqQkZsVER2ZDhWdkJuckxWZm1ROGhwVENBSWhVVG51LzIKSFNRdDFnZnBJUC9nL3dKUE5veUlQNmM4azd3dERFVHpVS0xvRVRFbmdTU0cyQlBICi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K"

** Note **: We have used this caBundle on our VM's to be able to connect to our internal vault server properly with out any issues.
Kubernetes Version: 1.24+

**Comment by moolen** [comment]
Hey, can you share where the error originates? Where exactly do you see the error? In the logs? Is it a event on the ClusterSecretStore?

**Comment by ramakrishnateja** [comment]
Yes its on the events

Events:
  Type     Reason                 Age                    From                  Message
  ----     ------                 ----                   ----                  -------
  Warning  InvalidProviderConfig  3m3s (x227 over 2d9h)  cluster-secret-store  unable to log in to auth method: unable to log in with app role auth: Put "https://vault-us.xxxinternal.com/v1/auth/approle/login": x509: certificate signed by unknown authority

**Comment by moolen** [comment]
Thanks, i see. I assume this is just an issue with the store validation, i'll look into it!

**Comment by moolen** [comment]
Yeah, you can try to start the controller with `--enable-flood-gate=false`.

**Comment by ramakrishnateja** [comment]
> Yeah, you can try to start the controller with `--enable-flood-gate=false`.

Perfect.Thanks. Will try.

**Comment by moolen** [comment]
Just looked at the code, the caBundle should be respected in the SecretStore validation step; please verify that the caBundle matches (it points to `couchbase-operator-admission CA` currently which doesn't look like vault related :thinking:)

**Comment by ramakrishnateja** [comment]
Let me re-verify. Thanks for the help
